### PR TITLE
Revert "Use cardano-config project from github."

### PR DESCRIPTION
**Pull request by tmcgilchrist**
This reverts commit 441115e3fe023144f633dd3e70c657738eca6024.

Further it introduces a new package `cardano-git-rev` to better capture what this code does.
Namely providing a mechanism for both `cabal` and `nix` to provide a git sha in the `--version` flag for `cardano-node`.

### CODE REVIEW

File: cardano-config/README.md
Comment (angerman) [inline cardano-config/README.md:13]: I don't follow.

Comment (angerman) [reply cardano-config/README.md:13]: We know this doesn't work with `nix`, though I still think we should be able to improve this with nix, and yes cross compilation is exclusive through nix, so this can't work,. It's in the `cabal` section though, so I'm fairly confused.

Comment (tmcgilchrist) [reply cardano-config/README.md:13]: Perhaps my wording isn't descriptive enough here and I can drop the details about cross-compiling.
The short version is `cabal` builds require `git` and a git checkout while building to get an accurate git sha. 

It took me a while to discover how both nix and cabal, get the right git sha patched into the final executable, I wanted to capture that knowledge somewhere. 

https://github.com/input-output-hk/cardano-node/issues/3440

File: cardano-config/LICENSE
Comment (Jimbo4350) [inline cardano-config/LICENSE:190]: Should be 2022

File: cardano-config/NOTICE
Comment (Jimbo4350) [inline cardano-config/NOTICE:1]: 2022

File: cardano-config/src/Cardano/Config/Git/RevFromGit.hs
Comment (angerman) [inline cardano-config/src/Cardano/Config/Git/RevFromGit.hs:25]: Why are we reverting this and not incorporate the fixes from `cardano-config`? This code here is clearly broken with some `process` libraries. If `git` is not found, this does not necessarily result in a `isDoesNotExistErrorType` but some other exception that is thrown. It's fairly annoying that the process lirbary got changed to support spawn, and now has a different behaviour. But this code is sadly broken as it is.

Comment (angerman) [reply cardano-config/src/Cardano/Config/Git/RevFromGit.hs:25]:
```
            # src/Cardano/Config/Git/Rev.hs:33:35: error:
            #     • Exception when trying to run compile-time code:
            #         git: readCreateProcessWithExitCode: posix_spawn_file_actions_adddup2(child_end): invalid argument (Invalid argument)
            #       Code: gitRevFromGit
            #     • In the untyped splice: $(gitRevFromGit)
            #    |
            # 33 |         fromGit = T.strip (T.pack $(gitRevFromGit))
            #    |
            ```

Comment (angerman) [reply cardano-config/src/Cardano/Config/Git/RevFromGit.hs:25]: (And similar exceptions...)

File: cardano-git-rev/src/Cardano/Config/Git/RevFromGit.hs
Comment (erikd) [inline cardano-git-rev/src/Cardano/Config/Git/RevFromGit.hs:1]: Maybe this module name should change?

File: cardano-git-rev/cardano-git-rev.cabal
Comment (erikd) [inline cardano-git-rev/cardano-git-rev.cabal:3]: I think I would just have called it `cardano-gitrev` .

File: cardano-config/src/Cardano/Config/Git/RevFromGit.hs
Comment (tmcgilchrist) [reply cardano-config/src/Cardano/Config/Git/RevFromGit.hs]: Copied the code fix across from https://github.com/input-output-hk/cardano-config/pull/4

File: cardano-git-rev/cardano-git-rev.cabal
Comment (Jimbo4350) [reply cardano-git-rev/cardano-git-rev.cabal:3]: @tmcgilchrist Does it have to be a separate package or can it be a module?

Comment (tmcgilchrist) [reply cardano-git-rev/cardano-git-rev.cabal:3]: It's easier to keep this as a small independent package since it's referred to in a number of places. Finding the root of all those to place it was used and putting the code there ended up being the worse option.

File: cardano-git-rev/src/Cardano/Config/Git/RevFromGit.hs
Comment (newhoggy) [reply cardano-git-rev/src/Cardano/Config/Git/RevFromGit.hs:1]: I agree.  The module name should change.

Comment (tmcgilchrist) [reply cardano-git-rev/src/Cardano/Config/Git/RevFromGit.hs]: Changed the module name.

**Review by newhoggy — APPROVED** [review]

File: cardano-git-rev/src/Cardano/Git/RevFromGit.hs
Comment (Jimbo4350) [inline cardano-git-rev/src/Cardano/Git/RevFromGit.hs:2]: So we format the export list as follows:
```
module Cardano.Git.RevFromGit 
  ( gitRevFromGit
  ) where
```
You can refer to `Export Lists` in the `STYLE.md`

File: cardano-git-rev/src/Cardano/Git/RevFromGit.hs
Comment (Jimbo4350) [inline cardano-git-rev/src/Cardano/Git/RevFromGit.hs:15]: We also indent with 2 spaces.

File: cardano-node/src/Cardano/Node/Configuration/Logging.hs
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Node/Configuration/Logging.hs:2]: :+1:

**Review by Jimbo4350 — APPROVED** [review]
LGTM :+1:. Approving with the caveat that the small style changes will be made before merging.

File: cardano-git-rev/src/Cardano/Git/RevFromGit.hs
Comment (tmcgilchrist) [reply cardano-git-rev/src/Cardano/Git/RevFromGit.hs]: Cheers, I didn't know about the style guide. 👍🏻

**Review by denisshevchenko — APPROVED** [review]

**Pull request merged**
